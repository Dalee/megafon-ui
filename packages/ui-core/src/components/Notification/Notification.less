@import '~styles/base';

@b: e('.mfui-notification');

@{b} {
    &__container {
        position: relative;
        max-width: 688px;
        padding: 12px 24px 12px 16px;
        border-width: 1px;
        border-style: solid;
        border-left-width: 8px;
        display: flex;
    }

    &__icon {
        height: 20px;
        min-width: 20px;
        width: 20px;
        margin-right: 16px;
        top: 2px;
        position: relative;
    }

    &_type_success {
        @{b} {
            &__container {
                border-color: @green;
            }

            &__icon {
                fill: @green;
            }
        }
    }

    &_type_warning {
        @{b} {
            &__container {
                border-color: @mandarin;
            }

            &__icon {
                fill: @mandarin;
            }
        }
    }

    &_type_error {
        @{b} {
            &__container {
                border-color: @fury;
            }

            &__icon {
                fill: @fury;
            }
        }
    }

    &_short {
        @{b}__container {
            border: none;
            max-width: 456px;
            padding-top: 4px;
            padding-bottom: 4px;
            padding-left: 24px;
        }

        @{b}__icon {
            fill: @clearWhite;
        }

        @{b}__text {
            color: @clearWhite;
        }
    }

    &_short&_type_success {
        @{b} {
            &__container {
                background-color: @green;
            }
        }
    }

    &_short&_type_warning {
        @{b} {
            &__container {
                background-color: @mandarin;
            }
        }
    }

    &_short&_type_error {
        @{b} {
            &__container {
                background-color: @fury;
            }
        }
    }
}
