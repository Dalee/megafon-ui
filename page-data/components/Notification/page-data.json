{"componentChunkName":"component---packages-ui-core-src-components-notification-notification-mdx","path":"/components/Notification","result":{"pageContext":{"frontmatter":{"name":"Notification","menu":"Компоненты","route":"components/Notification"},"entry":{"id":"df0ed0cc891e4618718a6b0a31cabbd0","filepath":"packages/ui-core/src/components/Notification/Notification.mdx","fullpath":"/home/runner/work/megafon-ui/megafon-ui/packages/ui-core/src/components/Notification/Notification.mdx","route":"components/Notification","slug":"packages-ui-core-src-components-notification-notification","name":"Notification","menu":"Компоненты","headings":[{"slug":"notification","depth":1,"value":"Notification"},{"slug":"properties","depth":2,"value":"Properties"},{"slug":"basic-types","depth":2,"value":"Basic types"},{"slug":"short-types","depth":2,"value":"Short types"}]}}}}