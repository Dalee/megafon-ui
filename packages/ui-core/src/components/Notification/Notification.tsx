import React from 'react';
import * as PropTypes from 'prop-types';
import './Notification.less';
import Paragraph from 'components/Paragraph/Paragraph';
import cnCreate from 'utils/cnCreate';
import AttentionIcon from 'icons/System/24/Info_24.svg';
import AttentionInvertIcon from 'icons/System/24/Info_invert_24.svg';
import CheckedIcon from 'icons/System/24/Checked_24.svg';

export type Props = {
    /** Types of notification */
    type?: 'success' | 'error' | 'warning';
    /** Short size of notification */
    isShort?: boolean;
    children: React.ReactNode;
};

const cn = cnCreate('mfui-notification');
const Notification: React.FC<Props> = ({ type, isShort, children }) => {
    function renderIcon(): JSX.Element {
        if (type !== 'success' && isShort) {
            return <AttentionInvertIcon className={cn('icon')} />;
        }
        if (type !== 'success' && !isShort) {
            return <AttentionIcon className={cn('icon')} />;
        }

        return <CheckedIcon className={cn('icon')} />;
    }

    return (
        <div
            className={cn('', {
                type,
                short: isShort,
            })}
        >
            <div className={cn('container')}>
                {renderIcon()}
                <Paragraph
                    className={cn('text')}
                    hasMargin={false}
                >
                    {children}
                </Paragraph>
            </div>
        </div>
    );
};

Notification.propTypes = {
    type: PropTypes.oneOf(['success', 'error', 'warning']),
    isShort: PropTypes.bool,
    children: PropTypes.node,
};

export default Notification;
