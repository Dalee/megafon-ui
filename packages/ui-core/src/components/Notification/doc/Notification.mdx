---
name: Notification
menu: Компоненты
route: components/Notification
---

import { Props } from 'docz';
import Notification from '../Notification';
import Example from './Notification.example.mdx';

# Notification

<ImportExample name="Notification" />

<DoczTabs>
    <Example />
    <Props of={Notification} />
</DoczTabs>
