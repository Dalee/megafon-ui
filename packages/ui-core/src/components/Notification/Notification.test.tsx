import * as React from 'react';
import { shallow } from 'enzyme';
import Notification from './Notification';

describe('<Notification />', () => {
    it('it renders Notification without props', () => {
        const wrapper = shallow(
            <Notification>Some test text</Notification>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification with type', () => {
        const wrapper = shallow(
            <Notification type="success">Some test text</Notification>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders short Notification', () => {
        const wrapper = shallow(
            <Notification isShort>Some test text</Notification>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
