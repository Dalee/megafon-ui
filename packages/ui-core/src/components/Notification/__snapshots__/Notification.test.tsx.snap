// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> it renders Notification with type 1`] = `
<div
  className="mfui-notification mfui-notification_type_success"
>
  <div
    className="mfui-notification__container"
  >
    <test-file-stub
      className="mfui-notification__icon"
    />
    <Paragraph
      className="mfui-notification__text"
      hasMargin={false}
    >
      Some test text
    </Paragraph>
  </div>
</div>
`;

exports[`<Notification /> it renders Notification without props 1`] = `
<div
  className="mfui-notification"
>
  <div
    className="mfui-notification__container"
  >
    <test-file-stub
      className="mfui-notification__icon"
    />
    <Paragraph
      className="mfui-notification__text"
      hasMargin={false}
    >
      Some test text
    </Paragraph>
  </div>
</div>
`;

exports[`<Notification /> it renders short Notification 1`] = `
<div
  className="mfui-notification mfui-notification_short"
>
  <div
    className="mfui-notification__container"
  >
    <test-file-stub
      className="mfui-notification__icon"
    />
    <Paragraph
      className="mfui-notification__text"
      hasMargin={false}
    >
      Some test text
    </Paragraph>
  </div>
</div>
`;
